statsd/telemetry_test.go

```diff
@@ -78,15 +155,25 @@ func testTelemetry(t *testing.T, telemetry *telemetryClient, expectedMetrics map
 	}
 	metrics := telemetry.flush()
 
-	assert.Equal(t, len(expectedMetrics), len(metrics))
-	for _, m := range metrics {
-		expectedValue, found := expectedMetrics[m.name]
-		assert.True(t, found, fmt.Sprintf("Unknown metrics: %s", m.name))
+	require.Equal(t, len(expectedMetrics), len(metrics), fmt.Sprintf("expected:\n%v\nactual:\n%v", expectedMetrics, metrics))
+
+	sort.Sort(metricSorted(metrics))
+	sort.Sort(metricSorted(expectedMetrics))
 
-		assert.Equal(t, expectedValue, m.ivalue, fmt.Sprintf("wrong ivalue for '%s'", m.name))
-		assert.Equal(t, count, m.metricType, fmt.Sprintf("wrong metricTypefor '%s'", m.name))
-		assert.Equal(t, expectedTelemetryTags, m.tags, fmt.Sprintf("wrong tags for '%s'", m.name))
-		assert.Equal(t, float64(1), m.rate, fmt.Sprintf("wrong rate for '%s'", m.name))
+	for idx := range metrics {
+		m := metrics[idx]
+
+		if idx >= len(expectedMetrics) {
+			assert.Fail(t, fmt.Sprintf("missing metrics '%s' with tags '%v'", m.name, m.tags))
+			continue
+		}
```

Comment:
Should not happen because of https://github.com/DataDog/datadog-go/pull/175/files?diff=split&w=1#diff-c26e0df45fa892f2ea03bac21620205a43abd4b854e899364633ded9d1cdcb7bR158